### PR TITLE
Header for templates

### LINKED ISSUE #84: All generated file should mention they were generated
I suggest adding a header to each template, as a comment in the target language, to mention two things:
- that they are generated
- how to regenerate them
- that this is documented here on coq-community/templates.

Hence I suggest that the following gets added to all generated file assuming
1. `#` is the symbol for commenting in the target language of a given template
2. that `meta.yml` is the name of the meta data file
```
# This file was generated from `meta.yml`, please do not edit manually.
# Follow the instructions on https://github.com/coq-community/templates to regenerate.
```

In a second step, we could provide a generated file `./regenerate.sh` (with possible option `--nix`) in order to reperform the instructions from https://github.com/coq-community/templates without further arguments...

CC: @Zimmi48 @strub

### PR DESCRIPTION
**Pull request by CohenCyril**
Fixes #84

CC @Zimmi48

### CODE REVIEW

File: index.md.mustache [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
+[comment]: # (This file was generated from `{{ meta_file }}{{^ meta_file}}meta.yml{{/ meta_file}}`, please do not edit manually.)
+[comment]: # (Follow the instructions on https://github.com/coq-community/templates to regenerate.)
```
Comment (Zimmi48) [inline index.md.mustache:3]: Simply using `#` in there was not supported?

**Review by Zimmi48 — COMMENTED** [review]
Since there is a new field `meta_file`, it should be documented in `ref.yml`.

**Comment by CohenCyril** [comment]
> Since there is a new field `meta_file`, it should be documented in `ref.yml`.

Since `meta.yml` is fixed by `./generate.sh` there is no reason to make it a field/variable.

**Comment by palmskog** [comment]
This looks good to me, but unfortunately needs a rebase after the Nix changes.

**Review by Zimmi48 — APPROVED** [review]

**Comment by Zimmi48** [comment]
I've had to revert the header comment for `dune-project` because it's apparently not allowed by Dune. I don't know if we can add comments at some other place in the file.

**Pull request merged**